### PR TITLE
Remove unnecessary mutability

### PR DESCRIPTION
**Pull request by moOsama76**
# Objective

- Describe the objective or issue this PR addresses.
- If you're fixing a specific issue, say "Fixes #X".

## Solution

- Describe the solution used to achieve the objective above.

## Testing

- Did you test these changes? If so, how?
- Are there any parts that need more testing?
- How can other people (reviewers) test your changes? Is there anything specific they need to know?
- If relevant, what platforms did you test these changes on, and are there any important ones you can't test?

---

## Showcase

> This section is optional. If this PR does not include a visual change or does not add a new feature, you can delete this section.

- Help others understand the result of this PR by showcasing your awesome work!
- If this PR adds a new feature or public API, consider adding a brief pseudo-code snippet of it in action
- If this PR includes a visual change, consider adding a screenshot, GIF, or video
  - If you want, you could even include a before/after comparison!
- If the Migration Guide adequately covers the changes, you can delete this section

While a showcase should aim to be brief and digestible, you can use a toggleable section to save space on longer showcases:

<details>
  <summary>Click to view showcase</summary>

```rust
println!("My super cool code.");
```

</details>

## Migration Guide

> This section is optional. If there are no breaking changes, you can delete this section.

- If this PR is a breaking change (relative to the last release of Bevy), describe how a user might need to migrate their code to support these changes
- Simply adding new functionality is not a breaking change.
- Fixing behavior that was definitely a bug, rather than a questionable design choice is not a breaking change.

### CODE REVIEW

File: examples/games/breakout.rs [hunk 1]
```diff
@@ -323,15 +323,14 @@ fn move_paddle(
...
-    if keyboard_input.pressed(KeyCode::ArrowRight) {
-        direction += 1.0;
-    }
+    let direction = if keyboard_input.pressed(KeyCode::ArrowLeft) {
+        -1.0
+    } else if keyboard_input.pressed(KeyCode::ArrowRight) {
+        1.0
+    } else {
+        0.0
+    };
```
Comment (lewiszlw) [inline examples/games/breakout.rs:333]: The behavior looks changed when user pressed `ArrowLeft` and `ArrowRight` simultaneously.🤔

Comment (moOsama76) [reply examples/games/breakout.rs:333]: Yes, but I think pressing both buttons shouldn't make the player paddle stand still, it's not an important thing so feel free to merge or not merge it 😊

Comment (alice-i-cecile) [reply examples/games/breakout.rs:333]: They should definitely stand still: this is a common source of bugs when writing input handling code :)

**Comment by alice-i-cecile** [comment]
Closing as won't fix, but I appreciate you taking a look and trying to improve our examples :) This category of bug is what leads to things like "walking diagonally is faster", so it's important to model correctly here.

**Pull request closed without merging**